### ISSUE TITLE
sometimes can't jump to tp.file.cursor

### ISSUE THREAD

**Issue by domeniczz** [SilentVoid13/Templater#928]
**Plugin information (please complete the following information):**

-   OS: windows 11
-   Templater version: 1.16.0
-   Obsidian version: 1.0.3
-   Templater settings: tp.file.cursor

**Describe the bug**

`tp.file.cursor` can't jump cursor to the position

cursor will jump to the begining of the template

The above bug doesn't appear in every note. In short notes (with little words), `tp.file.cursor` works well, the bug will appear in bigger notes.

I don't know what triggers this bug, and I can't tell the word count limit, like over how many words will trigger the bug. Thus, I can't present a sample note.

**Additional context**

If any extra info is needed, I'm glad to present.

**Comment by domeniczz** [comment]
> 

Yes, cursor will jump to before the first character of the template

But only happens in large files (contains a lot syntax like lists, tables, html, code blocks), I don't know why.

I couldn't reproduce it in new files

**Comment by acook** [comment]
I've been able to reliably replicate this issue.

Not working:
- If the file is created via a "new file creation" trigger, the cursor will _not_ move and it will be in the title after the note opens. eg when right clicking a folder and clicking "New Note" in a folder that has a Template.
- If the current file has a template applied to it via the left toolbar Templater button, there will be no cursor in the note at all. I assume when clicking on the toolbar, focus never made its way back.
- Templates created via ctrl-p then going to "Create new note from template" using only the keyboard result in the title being highlighted, instead of the indicated cursor position (even if the title is set to something by the template).

Working:
- Templates created from a button using the Button plugin work just fine and position the cursor where you expect. Yay!
- Templates inserted via ctrl-p then going to "Insert Template modal" using only the keyboard work great!
- Templates applied via ctrl-p then going to "Replace templates in active file" using only the keyboard also are working no problem.

There are definitely more combinations to try, but this is what I have figured out so far.

Also linking in this older issue as it seems to be related: https://github.com/SilentVoid13/Templater/issues/397

**Comment by welpdx** [comment]
@acook 
Thank you very much for your testing! I've decided to do test as well based on your finding to verify and to also learn about the issue. 


Anyways, here's my results/findings:


|   | Method                                      | Image                                                                                                                                                                | Works?                                                                                                                                                                                               | Steps                                                                                                                                                                                                       | Note                                                                                        |
|---|---------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------|
| 1 | Obsidian Create New file                    | <details>                 <summary   style="font-size:20">img</summary>                 <img   src="https://i.imgur.com/7vCvaMi.png" alt="demo"   />      </details> | <details>                 <summary   style="font-size:20">Yes (Conditionally). Click to   see</summary>                 <img   src="https://i.imgur.com/BcNpBTw.gif" alt="demo"   />      </details> | 1. Obsidian Create New File.   <br/>      2. Let Folder Template do it's thing <br/>      3. Focus is at the title. *Press enter after renaming.* <br/>      4. `<%tp.file.cursor(0)%>`  is replaced  <br/> | - Press Enter after editing   Title <br/>      - Require: Folder Template set up            |
| 2 | Workspace Ribbon Button                     | <details>                 <summary   style="font-size:20">img</summary>                 <img   src="https://i.imgur.com/1I5kDmn.png" alt="demo"   />      </details> | <details>                 <summary   style="font-size:20">Yes (Conditionally). Click to   see</summary>                 <img   src="https://i.imgur.com/mIaxzCO.gif" alt="demo"   />      </details> | 1. Click on Templater Ribbon   Button. <br/>      2. Let Folder Template do it's thing <br/>      3. *Click on Tab to focus*. <br/>      4. `<%tp.file.cursor(0)%>`  is   replaced  <br/>                   | - Click on tab to focus to the   jumped cursor <br/>      - Require: Folder Template set up |
| 3 | Templater: Create New Note From Template    | <details>                 <summary   style="font-size:20">img</summary>                 <img   src="https://i.imgur.com/a5ONNbp.png" alt="demo"   />      </details> | Yes. <br/> Same as \#\1.   Press Enter at Title                                                                                                                                                      | See 1st row                                                                                                                                                                                                 |  - Press Enter after editing Title                                                          |
| 4 | Templater: Insert Template modal            | <details>                 <summary   style="font-size:20">img</summary>                 <img   src="https://i.imgur.com/ZHSDsHk.png" alt="demo"   />      </details> | Yes                                                                                                                                                                                                  |                                                                                                                                                                                                             |                                                                                             |
| 5 | Templater: Replace templates in active file | <details>                 <summary   style="font-size:20">img</summary>                 <img   src="https://i.imgur.com/SgpP51K.png" alt="demo"   />      </details> | Yes                                                                                                                                                                                                  |                                                                                                                                                                                                             |                                                                                             |
|   | Other                                       |                                                                                                                                                                      |                                                                                                                                                                                                      |                                                                                                                                                                                                             |                                                                                             |
| 6 | Using Button Plugin                         |                                                                                                                                                                      |                                                                                                                                                                                                      |                                                                                                                                                                                                             |                                                                                             |

So, for me on windows, it seems like `tp.file.cursor(0)` does work as intended. For some you just need to press enter from the title or click the tab to focus in the document (albeit this one is a bit annoying).

**Comment by acook** [comment]
Interesting! Thanks for looking into it. 

I am on Linux/X11/Budgie, not sure if the cursor behaviour is any different on that setup or not.

1. Confirmed, the cursor jumps appropriately after enter!
2. Confirmed, if I click _only_ the tab, the cursor does indeed jump to the correct location.
3. As 1, after enter.
4. :+1: 
5. :+1: 
6. :+1: 

Looks like everything is the same, which is a relief.

I think the trouble is that Obsidian acts differently in each scenario, like the difference between 1 and 3 is that the title either has a cursor or has the title fully selected too, so it's harder to hook into I assume. This could be mentioned in documentation and I think that is all the fix most people would need. (it solves my use case at least)

Probably scenario 2 is the one that is the most vexsome for people, so that is the one I would vote gets attention when someone has time. It would involve telling Obsidian to change focus to the tab, not sure how troublesome that would be.

**Comment by iamfeiyu** [comment]
> 

是的 v1.16.0

**Comment by xiaojie0254** [comment]
I repeated the same question

**Comment by mjs271** [comment]
So this one's been vexing me for a bit, but I've run down what I think are some related causes.

- I believe it's partially related to 410c3a886e117e6a79849190aa0e9a9e64e56608 with the commit message:
> fix(cursorjumper): fix issues with jump cursor stealing focus away from the 'rename file' text field
> The cursor jumper should not override the editor focus. This was interfering with the "create new file" flow. Focus was being stolen from the rename file field and the cursor moves to the first jump location preemptively.
- I would guess that this leads to the behavior some users experience where the cursor starts in the title bar when the template employs either `tp.file.move()` or `tp.file.rename()`. Namely, my experience is:
  - Upon Templater completion, the cursor is in the note title field (the title I've provided).
  - Pressing <kbd> tab </kbd> jumps me to `tp.file.cursor()` or `tp.file.cursor(1)`, etc.
    - _Note_ that my keybinding for the jump may not be standard--honestly can't remember 🙃
- My hypothesis is that the above commit, preventing the `cursorjumper` from stealing focus from the "_'rename file' text field_," is interfering with the way a user would prefer it to work _**in the case that they have already renamed the file**_.
- _Note_, also, that I am employing Templater's **_Folder Templates_** functionality to auto-trigger the template on new file creation.

### Possible Fix

My initial, naive thought is that a string comparison could be done to determine whether the file has been given a non-default name and use this to control whether the cursor starts in the title or at the first `tp.file.cursor()`. E.g.,
```
if note.title != "Untitled"; then jump_to_first()...
```
Although, perhaps Obsidian provides a more clever way to do this.

Unfortunately, I am not experienced writing typescript/javascript nor am I very familiar with the Obsidian API, so it would take a lot for me to work out a fix (doesn't mean I won't eventually get motivated and give it a shot).

So... anyone up for looking into it? I'm happy to lend a hand if it's helpful.

#### Here's a minimal reproducer, in case that's helpful.
```markdown
# New Template Note

<%*
let noteName = await tp.system.prompt("What is the name of the note?")
await tp.file.move("noteFolder/" + noteName)
let dueDate = await tp.user.parseDueDate_NL(tp)
let reminderDate = await tp.date.now("YYYY-MM-DD", -7, dueDate, "YYYY-MM-DD")
_%>
---
due_date: <% dueDate %>
date_added: <% tp.file.creation_date("YYYY-MM-DD") %>
---

- [ ] #task <% noteName %> ⏰ <% reminderDate %> 📅 <% dueDate %>
---

## Notes

- <% tp.file.cursor() %>
```

**Comment by Zachatoo** [comment]
I'm adding a "Jump to note content after creating note" setting, when enabled, will jump your cursor to the note content without having to tab into it.

<img width="888" height="776" alt="Screenshot 2026-06-05 at 10 05 48 PM" src="https://github.com/user-attachments/assets/5a5b7074-66b9-4f63-b266-d272141d2d7e" />

This is in the Templater 2.22.0 release, which is only available on Obsidian 1.13.0+
